### PR TITLE
tests: use dir_helpers on repo

### PR DESCRIPTION
**Pull request by mroutis**

### CODE REVIEW

File: dvc/api.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ def get_url(path, repo=None, rev=None, remote=None):
     """Returns an url of a resource specified by path in repo"""
     with _make_repo(repo, rev=rev) as _repo:
         abspath = os.path.join(_repo.root_dir, path)
-        out, = _repo.find_outs_by_path(abspath)
+        (out,) = _repo.find_outs_by_path(abspath)
```
Comment (mroutis) [inline dvc/api.py]: black added this stuff

File: setup.py [hunk 2]
```diff
@@ -141,7 +141,7 @@ def run(self):
 ]
 
 if (sys.version_info) >= (3, 6):
-    tests_requirements.append("black==19.3b0")
+    tests_requirements.append("black==19.10b0")
```
Comment (mroutis) [inline setup.py]: black was not _working_ with Python 3.7.5, so I updated it to the latest version

File: tests/unit/test_repo.py [hunk 3]
```diff
@@ -1,13 +1,76 @@
...
 
-class TestIsDvcInternal(TestDvc):
-    def test_return_false_on_non_dvc_file(self):
-        path = os.path.join("path", "to-non-.dvc", "file")
-        self.assertFalse(self.dvc.is_dvc_internal(path))
 
-    def test_return_true_on_dvc_internal_file(self):
-        path = os.path.join("path", "to", ".dvc", "file")
-        self.assertTrue(path)
+def test_destroy(tmp_dir, dvc):
```
Comment (mroutis) [inline tests/unit/test_repo.py]: moved `.destroy`, `.collect`, `.status` to the unit test module. the reason behind it is that we are testing the behavior of a single unit, in this case, a single method.

File: tests/unit/test_repo.py [hunk 4]
```diff
@@ -1,13 +1,76 @@
...
+    tmp_dir.dvc_gen({"dir": {"file": "lorem", "subdir/file": "ipsum"}})
+
+    dvc.destroy()
+
+    # Remove all the files related to DVC
+    assert not (tmp_dir / ".dvc").exists()
+    assert not (tmp_dir / "file.dvc").exists()
+    assert not (tmp_dir / "dir.dvc").exists()
+
+    # Leave the rest of the files
```
Comment (mroutis) [inline tests/unit/test_repo.py]: could we write less assertions and cover all the cases?

File: tests/unit/test_repo.py [hunk 5]
```diff
@@ -1,13 +1,76 @@
...
+
+    # Make sure that data was unprotected after `destroy`
+    assert not System.is_symlink(fspath(tmp_dir / "foo"))
+    assert not System.is_symlink(fspath(tmp_dir / "file"))
+    assert not System.is_symlink(fspath(tmp_dir / "dir"))
+    assert not System.is_symlink(fspath(tmp_dir / "dir" / "file"))
+    assert not System.is_symlink(fspath(tmp_dir / "dir" / "subdir" / "file"))
+
+
+def test_collect(tmp_dir, scm, dvc, run_copy):
```
Comment (mroutis) [inline tests/unit/test_repo.py]: didn't get why we are creating branches, why not doing the following:

```python
def test_collect(tmp_dir, dvc, run_copy):
    def collect_outs(*args, **kwargs):
        return [
            str(out.path_info)
            for stage in dvc.collect(*args, **kwargs)
            for out in stage.outs
        ]

    tmp_dir.dvc_gen("foo", "foo")
    run_copy("foo", "bar")

    assert ["foo", "bar"] == collect_outs("bar.dvc", with_deps=True)

    run_copy("bar", "buzz")

    assert ["foo", "bar", "buzz"] == collect_outs("buzz.dvc", with_deps=True)
    assert ["buzz"] == collect_outs("buzz.dvc", with_deps=False)
```

**Comment by efiop** [comment]
@mroutis Please check tests, they are failing.

**Comment by mroutis** [comment]
@efiop , thanks for the notice

**Comment by efiop** [comment]
@mroutis tests are still failing

File: tests/unit/test_repo.py (hunk 3 above)
Comment (Suor) [reply tests/unit/test_repo.py]: Still look like functional tests to me. We recreate the whole scenario and call a method, which is a mirror of cli command.

File: tests/unit/test_repo.py (hunk 4 above)
Comment (Suor) [reply tests/unit/test_repo.py]: We can use a for loop :)

File: tests/unit/test_repo.py (hunk 5 above)
Comment (Suor) [reply tests/unit/test_repo.py]: No idea) May look at blame and ask whoever added this. But, again, this migration is not about unittests, so I suggest just revert all unittest changes for now. Or we might stuck with this, which is not good.

File: tests/unit/test_repo.py (hunk 3 above)
Comment (mroutis) [reply tests/unit/test_repo.py]: got it, will bring them back

File: tests/unit/test_repo.py (hunk 5 above)
Comment (mroutis) [reply tests/unit/test_repo.py]: @efiop , could you shade some light?

File: setup.py (hunk 2 above)
Comment (efiop) [reply setup.py]: @mroutis Need to update https://github.com/iterative/dvc/blob/master/.restyled.yaml too

Comment (mroutis) [reply setup.py]: @efiop , looks like restyled doesn't support black `19.10`:

> stderrUnable to find image 'restyled/restyler-black:v19.10b0' locally

File: tests/unit/test_repo.py (hunk 3 above)
Comment (mroutis) [reply tests/unit/test_repo.py]: @Suor , for future references, how do you make the distinction between functional and unit tests?

File: setup.py (hunk 2 above)
Comment (mroutis) [reply setup.py]: https://github.com/restyled-io/restylers/pull/50

It's going to take a while... I guess I'll create an issue and handle this in another PR. Meanwhile, I'll work with Python 3.6

Comment (efiop) [reply setup.py]: works fine for me on 3.7.3

File: tests/func/test_repo.py [hunk 6]
```diff
@@ -1,71 +1,60 @@
...
-        self._check("master", self.BAR, True, [[self.FOO], [self.BAR]])
-        self._check(
-            "new_branch", "buzz", True, [[self.FOO], [self.BAR], ["buzz"]]
+from dvc.system import System
+from dvc.utils.compat import fspath
+
+
+def test_destroy(tmp_dir, dvc):
+    dvc.config.set("cache", "type", "symlink")
+
```
Comment (pared) [inline tests/func/test_repo.py]: `dvc.cache = Cache(dvc)`
setting cache type does not reload it,  and last check is always true

File: tests/func/test_repo.py [hunk 7]
```diff
@@ -1,71 +1,60 @@
...
+    def stages():
+        return set(stage.relpath for stage in Repo(str(tmp_dir)).stages)
 
-        stages = self.dvc.stages
-        self.assertEqual(4, len(stages))
+    tmp_dir.dvc_gen({"file": "a", "dir/file": "b", "dir/subdir/file": "c"})
 
-        self.create(DvcIgnore.DVCIGNORE_FILE, self.DATA_DIR)
+    assert stages() == {
+        os.path.join("file.dvc"),
```
Comment (pared) [inline tests/func/test_repo.py]: is there need for `os.path.join` here?

File: tests/func/test_repo.py [hunk 8]
```diff
@@ -1,71 +1,60 @@
...
-        result = self._check("new_branch", "buzz", False, [["buzz"]])
-        self.assertEqual([str(i) for i in result[0].deps], ["bar"])
 
+    tmp_dir.dvc_gen("foo", "foo")
+    run_copy("foo", "bar")
+    assert collect_outs("bar.dvc", with_deps=True) == {"foo", "bar"}
+
+    run_copy("bar", "buzz")
+    assert collect_outs("buzz.dvc", with_deps=True) == {"foo", "bar", "buzz"}
+    assert collect_outs("buzz.dvc", with_deps=False) == {"buzz"}
```
Comment (Suor) [inline tests/func/test_repo.py]: The "buzz" was added in a branch originally. And it was a collect with a branch assigned.

File: tests/func/test_repo.py [hunk 9]
```diff
@@ -1,71 +1,60 @@
...
-class TestIgnore(TestDvcGit):
-    def _stage_name(self, file):
-        return file + Stage.STAGE_FILE_SUFFIX
 
-    def test_should_not_gather_stage_files_from_ignored_dir(self):
-        ret = main(["add", self.FOO, self.BAR, self.DATA, self.DATA_SUB])
-        self.assertEqual(0, ret)
+def test_stages(tmp_dir, dvc):
+    def stages():
+        return set(stage.relpath for stage in Repo(str(tmp_dir)).stages)
```
Comment (Suor) [inline tests/func/test_repo.py]: `str` -> `fspath`

File: tests/unit/test_repo.py [hunk 10]
```diff
@@ -1,13 +1,6 @@
...
-    def test_return_false_on_non_dvc_file(self):
-        path = os.path.join("path", "to-non-.dvc", "file")
-        self.assertFalse(self.dvc.is_dvc_internal(path))
-
-    def test_return_true_on_dvc_internal_file(self):
-        path = os.path.join("path", "to", ".dvc", "file")
-        self.assertTrue(path)
+def test_is_dvc_internal(tmp_dir, dvc):
+    assert dvc.is_dvc_internal(fspath(tmp_dir / ".dvc" / "file"))
+    assert not dvc.is_dvc_internal("file")
```
Comment (Suor) [inline tests/unit/test_repo.py]: The test was quite different before:
- testing relative paths always
- testing `.dvc` not in the repo root (still considered internal, don't know why)
- testing tricky dirname "to-non-.dvc" (not sure if that is important)

File: tests/func/test_repo.py (hunk 7 above)
Comment (mroutis) [reply tests/func/test_repo.py]: no need, but just to keep it consistent :

File: tests/unit/test_repo.py (hunk 10 above)
Comment (mroutis) [reply tests/unit/test_repo.py]: @pared , where those intended? I'll leave them as it was before for now

File: tests/func/test_repo.py (hunk 8 above)
Comment (mroutis) [reply tests/func/test_repo.py]: I think it doesn't make sense to do the branching, but I'll do it just to not be a blocker for this PR

**Pull request merged**